elyra/pipeline/component_catalog.py

```diff
@@ -195,46 +292,122 @@ class ComponentCache(SingletonConfigurable):
-        for catalog in all_catalogs:
-            self.update_cache_for_catalog(catalog)
+    def update(self, catalog: Metadata, action: str):
+        """
+        Triggers an update of the component cache for the given catalog name.  If this is a non-server
+        process, the entry is written to the manifest file where it will be "processed" by the watchdog
+        and inserted into the component cache queue, otherwise we update the cache queue directly.
+        """
+        if self.is_server_process:
+            self.cache_queue.put((catalog, action))
+        else:
+            manifest: Dict[str, str] = self._load_manifest()
+            manifest[catalog.name] = action
+            self._update_manifest(manifest=manifest)
 
-    def update_cache_for_catalog(self, catalog: ComponentCatalogMetadata, operation: Optional[str] = None):
+    def _remove_all_manifest_files(self):
         """
-        Triggers an update of the component cache for the given ComponentCatalogMetadata instance.
+        Remove all existing manifest files in the Jupyter runtimes directory.
         """
-        self._cache_task_queue.put((catalog, operation))
+        manifest_files = Path(self.manifest_dir).glob('**/elyra-component-manifest-*.json')
+        for file in manifest_files:
+            os.remove(str(file))
+
+    def _load_manifest(self, filename: Optional[str] = None) -> Dict[str, str]:
+        """Read and return the contents of a manifest file.
 
-    def wait_for_all_cache_updates(self):
+        If 'filename' is not provided, this process's manifest file will be read.
+        """
+        filename = filename or self.manifest_filename
+        if not os.path.isfile(filename):
+            self.log.debug(f"Manifest file '{filename}' doesn't exist and will be created.")
+            return {}
+        with open(filename, 'r') as f:
+            manifest: Dict[str, str] = json.load(f)
+        self.log.debug(f"Reading manifest '{manifest}' from file '{filename}'")
+        return manifest
+
+    def _update_manifest(self, filename: Optional[str] = None, manifest: Optional[Dict[str, str]] = None) -> None:
+        """Update the manifest file with the given entry."""
+        filename = filename or self.manifest_filename
+        manifest = manifest or {}
+        self.log.debug(f"Updating manifest '{manifest}' to file '{filename}'")
+        with open(filename, 'w') as f:
+            json.dump(manifest, f, indent=2)
+
+    def wait_for_all_cache_tasks(self):
         """
         Block execution and wait for all tasks in the cache task update queue to complete.
         Primarily used for testing.
         """
-        self._cache_task_queue.join()
+        if self.is_server_process:
+            self.cache_queue.join()
 
     def get_all_components(self, platform: RuntimeProcessorType) -> List[Component]:
         """
         Retrieve all components from component catalog cache
         """
         components: List[Component] = []
```

Comment:
```suggestion
        components: List[Union[str, List[str]]] = []
```

Reply:
Huh my IDE doesn't yell at me for this (or the above case). Looking into it now. It should be returning `List[Component]` but maybe something is wrong with the `ComponentCacheType` typehint or something

Reply:
I think the type hint should actually be: `ComponentCacheType = Dict[str, Dict[str, Dict[str, Dict[str, Union[Component, str, List[str]]]]]]`. Making a note of it here so that I can change that after @kevin-bates pushes a few of his local commits.